topology/intel/dmic/dmic_macros.h

```diff
@@ -0,0 +1,281 @@
+/*
+  Copyright(c) 2021 Intel Corporation
+  All rights reserved.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of version 2 of the GNU General Public License as
+  published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St - Fifth Floor, Boston, MA 02110-1301 USA.
+  The full GNU General Public License is included in this distribution
+  in the file called LICENSE.GPL.
+*/
+
+#ifndef __DMIC_MACROS_H
+#define __DMIC_MACROS_H
+
+#define MIN(a, b) ({		\
```

Comment:
well here it is macro used only by the dmic code :) Anyway these are macros that could be used in generic way. I have similar thing in SSP directory -> I can try to unify those. It could be that at some point here's a new endpoint that can use these macros. It is just about the code partitioning and where to "hide" the files so the directory would look clean.